### PR TITLE
Support more labels per pr

### PR DESCRIPTION
**Pull request by AlexGhiondea**

### CODE REVIEW

File: tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs [hunk 1]
```diff
@@ -131,11 +134,33 @@ private string ToConfigString(CodeOwnerEntry entry)
...
+            if (File.Exists(fileOrUri))
+            {
+                return File.ReadAllText(fileOrUri);
+            }
+
+            // try to parse it as an Uri
+            Uri uri = new Uri(fileOrUri, UriKind.Absolute);
+            if (uri.Scheme.ToLowerInvariant() !="https")
+            {
+                throw new ArgumentException("Cannot download off non-https uris");
```
Comment (weshaggard) [inline tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs:148]: Why not?

File: tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs [hunk 2]
```diff
@@ -131,11 +134,33 @@ private string ToConfigString(CodeOwnerEntry entry)
...
+                return response.Content.ReadAsStringAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+            }
+        }
+
         private List<CodeOwnerEntry> ReadOwnersFromFile()
         {
             List<CodeOwnerEntry> entries = new List<CodeOwnerEntry>();
             string line;
-            using (StreamReader sr = new StreamReader(_codeownersFile))
+            using (StringReader sr = new StringReader(GetFileContents(_codeownersFile)))
```
Comment (weshaggard) [inline tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs:163]: Where is the multiple labels coming into play?

File: tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs [hunk 3]
```diff
@@ -131,11 +134,33 @@ private string ToConfigString(CodeOwnerEntry entry)
...
+        private string GetFileContents(string fileOrUri)
+        {
+            if (File.Exists(fileOrUri))
+            {
+                return File.ReadAllText(fileOrUri);
+            }
+
+            // try to parse it as an Uri
+            Uri uri = new Uri(fileOrUri, UriKind.Absolute);
+            if (uri.Scheme.ToLowerInvariant() !="https")
```
Comment (weshaggard) [inline tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs:146]: nit:
```suggestion
            if (uri.Scheme.ToLowerInvariant() !=" https")
```

File: tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs (hunk 2 above)
Comment (AlexGhiondea) [reply tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs:163]: From the empty string in the exclude entry in the payload. This is based on how the service works.

File: tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs [hunk 4]
```diff
@@ -131,11 +134,33 @@ private string ToConfigString(CodeOwnerEntry entry)
...
+        private string GetFileContents(string fileOrUri)
+        {
+            if (File.Exists(fileOrUri))
+            {
+                return File.ReadAllText(fileOrUri);
+            }
+
+            // try to parse it as an Uri
+            Uri uri = new Uri(fileOrUri, UriKind.Absolute);
+            if (uri.Scheme.ToLowerInvariant() !=" https")
```
Comment (weshaggard) [inline tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs:146]:
```suggestion
            if (uri.Scheme.ToLowerInvariant() != "https")
```

File: tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs [hunk 5]
```diff
@@ -31,9 +33,10 @@ public class PullRequestLabelFolderCapability : BaseCapability
           {
             ""label"": ""###label###"",
             ""pathFilter"": [ ###srcFolders### ],
-            ""exclude"": [ ]
+            ""exclude"": [ """" ]
```
Comment (weshaggard) [inline tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/PullRequestLabelFolderCapability.cs:36]: Might be worth adding a comment about why this empty string is doing.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**